src/entity_utils.cc

```diff
@@ -89,41 +92,51 @@ load_entities() {
 
                 auto stub_ptr = component_system_man.managers.get_stub(component->name, component).release();
 
-                entity.components.emplace_back(stub_ptr);
-
                 auto type_stub = dynamic_cast<type_component_stub*>(stub_ptr);
                 if (type_stub) {
                     entity.name = type_stub->name;
-
-                    found[component->name] = true;
                 }
 
-                auto render_stub = dynamic_cast<renderable_component_stub*>(stub_ptr);
-                if (render_stub) {
-                    found[component->name] = true;
+                auto deps = stub_ptr->get_dependencies();
+                dependencies[component->name] = {};
+                for (auto &&dep : deps) {
+                    dependencies[component->name][dep] = false;
                 }
 
-                auto pos_stub = dynamic_cast<relative_position_component_stub*>(stub_ptr);
-                if (pos_stub) {
-                    found[component->name] = true;
-                }
+                entity.components.emplace_back(stub_ptr);
+            }
 
-                auto physics_stub = dynamic_cast<physics_component_stub*>(stub_ptr);
-                if (physics_stub) {
-                    found[component->name] = true;
-                }
+            std::set<std::string> keys;
+            for (auto &&comp : dependencies) {
+                keys.insert(comp.first);
+            }
 
-                auto surf_stub = dynamic_cast<surface_attachment_component_stub*>(stub_ptr);
-                if (surf_stub) {
-                    found[component->name] = true;
+            for (auto &&comp : dependencies) {
+                for (auto &&dep : comp.second) {
+                    dep.second = keys.count(dep.first) != 0;
```

Comment:
Underneath all this awkward double-handling, isn't this the difference between a couple of different set unions?